vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -15,6 +15,8 @@ def initialize(occupancy)
       end
 
       def process
+        return unless user_can_purchase_secure_room?
```

Comment:
We should definitely test for the case of a facility operator who has accounts that could purchase. There's a very high probability of that.

For them, exiting the room shouldn't create a problem order.